### ISSUE TITLE
use the mobilenet_v1 as a backbone to train!

### ISSUE THREAD

**Issue by chrispolo** [Superlee506/Mask_RCNN#18]
Hello, @Superlee506 , when i changed the  resnet with mobilenet_v1 to train the humanpose keypoint, i met this problems.
 387/1000 [==========>...................] - ETA: 18:54 - loss: 1.1863 - rpn_class_loss: 0.5719 - rpn_bbox_loss: 0.5786 - mrcnn_class_loss: 0.0027 - mrcnn_bbox_loss: 0.0169 - keypoint_mrcnn_mask_loss: 0.0000e+00 - mrcnn_mask_loss: 0.0161
 388/1000 [==========>...................] - ETA: 18:52 - loss: 1.1860 - rpn_class_loss: 0.5718 - rpn_bbox_loss: 0.5785 - mrcnn_class_loss: 0.0027 - mrcnn_bbox_loss: 0.0169 - keypoint_mrcnn_mask_loss: 0.0000e+00 - mrcnn_mask_loss: 0.0160
 389/1000 [==========>...................] - ETA: 18:50 - loss: 1.1866 - rpn_class_loss: 0.5717 - rpn_bbox_loss: 0.5792 - mrcnn_class_loss: 0.0027 - mrcnn_bbox_loss: 0.0168 - keypoint_mrcnn_mask_loss: 0.0000e+00 - mrcnn_mask_loss: 0.0160
 390/1000 [==========>...................] - ETA: 18:48 - loss: 1.1870 - rpn_class_loss: 0.5716 - rpn_bbox_loss: 0.5799 - mrcnn_class_loss: 0.0027 - mrcnn_bbox_loss: 0.0168 - keypoint_mrcnn_mask_loss: 0.0000e+00 - mrcnn_mask_loss: 0.0160
 391/1000 [==========>...................] - ETA: 18:46 - loss: 1.1863 - rpn_class_loss: 0.5714 - rpn_bbox_loss: 0.5795 - mrcnn_class_loss: 0.0027 - mrcnn_bbox_loss: 0.0168 - keypoint_mrcnn_mask_loss: 0.0000e+00 - mrcnn_mask_loss: 0.0159
**the keypoint_mrcnn_mask_loss: 0.0000e+00**
I used the tesla M10 and p100 to train,also showed  met like this!
i will upload the files changed ,hope you can check it   at your convenience!
[mobilenet_humanpose.zip](https://github.com/Superlee506/Mask_RCNN_Humanpose/files/1969582/mobilenet_humanpose.zip)

**Comment by Superlee506** [comment]
@chrispolo I will check it later.

**Comment by chrispolo** [comment]
@Superlee506 ，Thx  very much!
After i trained some epochs:
the loss is :996/1000 [============================>.] - ETA: 7s - loss: 1.5868 - rpn_class_loss: 0.5630 - rpn_bbox_loss: 0.5673 - mrcnn_class_loss: 0.0016 - mrcnn_bbox_loss: 0.0196 - keypoint_mrcnn_mask_loss: 0.3959 - mrcnn_mask_loss: 0.0391
 997/1000 [============================>.] - ETA: 5s - loss: 1.5860 - rpn_class_loss: 0.5630 - rpn_bbox_loss: 0.5669 - mrcnn_class_loss: 0.0016 - mrcnn_bbox_loss: 0.0196 - keypoint_mrcnn_mask_loss: 0.3955 - mrcnn_mask_loss: 0.0391
 998/1000 [============================>.] - ETA: 3s - loss: 1.5855 - rpn_class_loss: 0.5630 - rpn_bbox_loss: 0.5670 - mrcnn_class_loss: 0.0016 - mrcnn_bbox_loss: 0.0196 - keypoint_mrcnn_mask_loss: 0.3951 - mrcnn_mask_loss: 0.0390
 999/1000 [============================>.] - ETA: 1s - loss: 1.5848 - rpn_class_loss: 0.5630 - rpn_bbox_loss: 0.5667 - mrcnn_class_loss: 0.0016 - mrcnn_bbox_loss: 0.0196 - keypoint_mrcnn_mask_loss: 0.3947 - mrcnn_mask_loss: 0.0390
1000/1000 [==============================] - 1796s 2s/step - loss: 1.5843 - rpn_class_loss: 0.5629 - rpn_bbox_loss: 0.5667 - mrcnn_class_loss: 0.0016 - mrcnn_bbox_loss: 0.0196 - keypoint_mrcnn_mask_loss: 0.3943 - mrcnn_mask_loss: 0.0389 - val_loss: 1.0052 - val_rpn_class_loss: 0.5905 - val_rpn_bbox_loss: 0.4145 - val_mrcnn_class_loss: 0.0000e+00 - val_mrcnn_bbox_loss: 0.0000e+00 - val_keypoint_mrcnn_mask_loss: 0.0000e+00 - val_mrcnn_mask_loss: 0.0000e+00
Epoch 60/200

but the val_ loss is still the 0.0000e+00, and i test the trained weight after 60 epoches
the result is 
![download](https://user-images.githubusercontent.com/37335192/39627083-1ec4d8c4-4fd7-11e8-8285-78292d02ca1e.png)
i can't see any keypoints. Hope you can check it later.

**Comment by chrispolo** [comment]
I also confused that how to use the maskrcnn to identity the keypoint of the human in your project,Can you explain how did you to achieve it  ? and if you can, you could recommend some blogs or project about this principle.
I'm sorry to bother you many times!  
Thx a lot  and hope your reply!

**Comment by Superlee506** [comment]
@chrispolo Sorry for my late reply.
one-hot vector of 56*56 is used as the keypoint label which is different from the original heatmap.
During testing, when we get the label of the output, we can change it into the index of height and width of 56*56 map with (h_index,w_index). Finally, the index can be resized into original image and we can get the final coordinates.

**Comment by chrispolo** [comment]
@Superlee506 , Thx  a lot！I will learn the one-hot encoding. How can i get the heatmaps of 17 keypoints in your project? and  i also  want to know whether the mobilenet as a backone in the project is useful?
I appreciate a lot for your guaidance.

**Comment by Superlee506** [comment]
@chrispolo "keypoint_mrcnn_mask" is also contained in the output of the model. If you want to inspect the model, you can refer to ["inspect_humanpose.ipynb"](https://github.com/Superlee506/Mask_RCNN_Humanpose/blob/master/inspect_humanpose.ipynb), and then you can run the sub-graph, to get the keypoint by it's name: "mrcnn_keypoint_mask_reshape"

**Comment by chrispolo** [comment]
@Superlee506 Thx  very much! I saw the inspect_humanpose.ipynb 
and notice the **Inspect some layers in build_fpn_keypoint_graph**
rpn = model.run_graph(inputs, [
    ("mrcnn_keypoint_mask_upsample_1", model.keras_model.get_layer("mrcnn_keypoint_mask_upsample_1").output),
    ("mrcnn_keypoint_mask_transpose", model.keras_model.get_layer("mrcnn_keypoint_mask_transpose").output),
    ("mrcnn_keypoint_mask_reshape", model.keras_model.get_layer("mrcnn_keypoint_mask_reshape").output),
])
But i don't know how to  visualize and get the heatmaps of the mrcnn_keypoint_mask_reshape.

**Comment by chrispolo** [comment]
@Superlee506 Sorry to bother you many times,i want to ask you again.Yesterday i try want to inspect the "mrcnn_keypoint_mask_reshape"
but i dont know to visualize the output of the mrcnn_keypoint_mask_reshape and inspect the heatmap of the 17 keypoints.
Thx a lot! Hope your reply!
